codegen/src/generator/rest_json.rs

```diff
@@ -115,6 +116,28 @@ impl GenerateProtocol for RestJsonGenerator {
     }
 }
 
+// Used to print the enum value rather than the status code and the canonical reason.
+// For codegen purposes, leaving existing StatusCode Display trait implementation intact.
+// StatusCode::Ok.to_string() prints "200 OK"
+// StatusCode::Ok.enum_as_string() prints "StatusCode::Ok"
+trait CodegenString {
+    fn enum_as_string(&self) -> String;
+}
+impl CodegenString for StatusCode {
+    fn enum_as_string(&self) -> String {
+        format!("StatusCode::{:?}", self)
+    }
+}
+
+fn http_code_to_status_code(code: Option<i32>) -> String {
+    match code {
+        Some(actual_code) => {
+            StatusCode::from_u16(actual_code as u16).enum_as_string()
+        }
+        None => "StatusCode::Ok".to_string(),
```

Comment:
I'm of the opinion that we should explode if we get an HTTP status code that we don't recognize. It seems an unlikely scenario, and one that would indicate that something Very Bad was happening.